xray/cli.go

```diff
@@ -274,6 +285,23 @@ func createServerDetailsWithConfigOffer(c *cli.Context) (*coreconfig.ServerDetai
 }
 
 func shouldIncludeVulnerabilities(c *cli.Context) bool {
-	// If no contex was provided by the user, no Violations will be triggered by Xray, so include general vulnerabilities in the command output
+	// If no context was provided by the user, no Violations will be triggered by Xray, so include general vulnerabilities in the command output
 	return c.String("watches") == "" && c.String("project") == "" && c.String("repo-path") == ""
 }
+
+func validateXrayContex(c *cli.Context) error {
+	contextFound := false
+	contextErorrMsg := "only one of the following flags can be supplied: --watches,--project or --repo-path."
+	if c.String("watches") != "" {
+		contextFound = true
+	}
+	if c.String("project") != "" && !contextFound {
+		contextFound = true
+	} else if c.String("project") != "" {
+		return errorutils.CheckError(errors.New(contextErorrMsg))
+	}
+	if c.String("repo-path") != "" && contextFound {
+		return errorutils.CheckError(errors.New(contextErorrMsg))
+	}
+	return nil
+}
```

Comment:
The above function can be simplified as follows:
```
func validateXrayContex(c *cli.Context) error {
	i := 0
	if c.String("watches") != "" {
		i++
	}
	if c.String("project") != "" {
		i++
	}
	if c.String("repo-path") != "" {
		i++
	}
	if u > 1 {
		return errorutils.CheckError(errors.New("only one of the following flags can be supplied: --watches,--project or --repo-path"))
	}
	return nil
}
```

Also, let's add a unit test for this function. See examples for testing flags in ```cli_test.go```.